pkg/operator/controller/controller.go

```diff
@@ -0,0 +1,864 @@
+	if err := r.checkUpgrade(logger, cr); err != nil {
+		return reconcile.Result{}, err
+	}
+
+	resources, err := r.getAllResources(cr)
+	if err != nil {
+		return reconcile.Result{}, err
+	}
+
+	var allErrors []error
+	for _, desiredRuntimeObj := range resources {
+		desiredMetaObj := desiredRuntimeObj.(metav1.Object)
+		currentRuntimeObj := newDefaultInstance(desiredRuntimeObj)
+
+		key := client.ObjectKey{
+			Namespace: desiredMetaObj.GetNamespace(),
+			Name:      desiredMetaObj.GetName(),
+		}
+		err = r.client.Get(context.TODO(), key, currentRuntimeObj)
+
+		if err != nil {
+			if !errors.IsNotFound(err) {
+				return reconcile.Result{}, err
+			}
+
+			setLastAppliedConfiguration(desiredMetaObj)
+			setLabel(createVersionLabel, r.operatorArgs.OperatorVersion, desiredMetaObj)
+
+			if err = controllerutil.SetControllerReference(cr, desiredMetaObj, r.scheme); err != nil {
+				return reconcile.Result{}, err
+			}
+
+			currentRuntimeObj = desiredRuntimeObj.DeepCopyObject()
+			logger.Info("Resource creating",
+				"namespace", desiredMetaObj.GetNamespace(),
+				"name", desiredMetaObj.GetName(),
+				"type", fmt.Sprintf("%T", desiredMetaObj))
+
+			if err = r.client.Create(context.TODO(), currentRuntimeObj); err != nil {
+				logger.Error(err, "")
+				allErrors = append(allErrors, err)
+				continue
+			}
+
+			logger.Info("Resource created",
+				"namespace", desiredMetaObj.GetNamespace(),
+				"name", desiredMetaObj.GetName(),
+				"type", fmt.Sprintf("%T", desiredMetaObj))
+		} else {
+			currentRuntimeObjCopy := currentRuntimeObj.DeepCopyObject()
+			currentMetaObj := currentRuntimeObj.(metav1.Object)
+
+			if !r.isMutable(currentRuntimeObj) {
+				setLastAppliedConfiguration(desiredMetaObj)
+
+				// overwrite currentRuntimeObj
+				currentRuntimeObj, err = mergeObject(desiredRuntimeObj, currentRuntimeObj)
+				if err != nil {
+					return reconcile.Result{}, err
+				}
+				currentMetaObj = currentRuntimeObj.(metav1.Object)
+			}
+
+			if !reflect.DeepEqual(currentRuntimeObjCopy, currentRuntimeObj) {
+				logJSONDiff(logger, currentRuntimeObjCopy, currentRuntimeObj)
+
+				setLabel(updateVersionLabel, r.operatorArgs.OperatorVersion, currentMetaObj)
+
+				if err = r.client.Update(context.TODO(), currentRuntimeObj); err != nil {
+					logger.Error(err, "")
+					allErrors = append(allErrors, err)
+					continue
+				}
+
+				logger.Info("Resource updated",
+					"namespace", desiredMetaObj.GetNamespace(),
+					"name", desiredMetaObj.GetName(),
+					"type", fmt.Sprintf("%T", desiredMetaObj))
+			} else {
+				logger.V(3).Info("Resource unchanged",
+					"namespace", desiredMetaObj.GetNamespace(),
+					"name", desiredMetaObj.GetName(),
+					"type", fmt.Sprintf("%T", desiredMetaObj))
+			}
+		}
+	}
+
+	if len(allErrors) > 0 {
+		return reconcile.Result{}, fmt.Errorf("reconcile encountered %d errors", len(allErrors))
```

Comment:
no need to update the VMImportConfig status with the errors ?

Reply:
we could recover from it.